src/app/exhibitor/_components/CompanySubmissionPopover.tsx

```diff
@@ -0,0 +1,172 @@
+"use client"
+import { sendToSlack } from "@/app/exhibitor/actions"
+import { Button } from "@/components/ui/button"
+import { env } from "@/env"
+import * as Popover from "@radix-ui/react-popover"
+import { Headset, X } from "lucide-react"
+import { useEffect, useRef, useState } from "react"
+import ReCAPTCHA from "react-google-recaptcha"
+
+export function CompanySubmissionPopover() {
+	const recaptcha = useRef<any>()
+	const [formData, setFormData] = useState({
+		name: "",
+		company: "",
+		email: "",
+		message: ""
+	})
+	const [isVerified, setIsVerified] = useState(false)
+	const [formedFilled, setFormFilled] = useState(false)
+	const [isOpen, setIsOpen] = useState(false)
+
+	useEffect(() => {
```

Comment:
Always give it a second round of thought before using useEffect, it doesn't have to be bad but it makes the code a little harder to read, in this case you could write the same code like this:

```tsx
const [isVerified, setIsVerified] = useState(false)
	//const [formedFilled, setFormFilled] = useState(false)
	const [isOpen, setIsOpen] = useState(false)

    const formFilled = useMemo(() => 
		formData.name !== "" &&
		formData.email !== "" &&
		formData.company !== "" &&
		formData.message !== "" &&
		isVerified, 
        [formData, isVerified]
    )

/* 	useEffect(() => {
		if (
			formData.name !== "" &&
			formData.email !== "" &&
			formData.company !== "" &&
			formData.message !== "" &&
			isVerified
		) {
			setFormFilled(true)
		} else {
			setFormFilled(false)
		}
	}, [formData, isVerified]) */
```